src/components/datagrid/body/header/data_grid_header_cell.tsx

```diff
@@ -180,16 +185,75 @@ export const EuiDataGridHeaderCell: FunctionComponent<EuiDataGridHeaderCellProps
           }
           isOpen={isPopoverOpen}
           closePopover={() => setIsPopoverOpen(false)}
+          {...popoverArrowNavigationProps}
         >
-          <div>
-            <EuiListGroup
-              listItems={columnActions}
-              gutterSize="none"
-              data-test-subj={`dataGridHeaderCellActionGroup-${id}`}
-            />
-          </div>
+          <EuiListGroup
+            listItems={columnActions}
+            gutterSize="none"
+            data-test-subj={`dataGridHeaderCellActionGroup-${id}`}
+          />
         </EuiPopover>
       )}
     </EuiDataGridHeaderCellWrapper>
   );
 };
+
+/**
+ * Add keyboard arrow navigation to the cell actions popover
+ * to match the UX of the rest of EuiDataGrid
+ */
+export const usePopoverArrowNavigation = () => {
+  const popoverPanelRef = useRef<HTMLElement | null>(null);
+  const actionsRef = useRef<FocusableElement[] | undefined>(undefined);
+  const panelRef = useCallback((ref) => {
+    popoverPanelRef.current = ref;
+    actionsRef.current = ref ? tabbable(ref) : undefined;
+  }, []);
+
+  const onKeyDown = useCallback((e: React.KeyboardEvent) => {
+    if (e.key !== keys.ARROW_DOWN && e.key !== keys.ARROW_UP) return;
+    if (!actionsRef.current?.length) return;
+
+    e.preventDefault();
+
+    const initialState = document.activeElement === popoverPanelRef.current;
+    const currentIndex = !initialState
+      ? actionsRef.current.findIndex((el) => document.activeElement === el)
+      : -1;
+    const lastIndex = actionsRef.current.length - 1;
+
+    let indexToFocus: number;
+    if (initialState) {
+      if (e.key === keys.ARROW_DOWN) {
+        indexToFocus = 0;
+      } else if (e.key === keys.ARROW_UP) {
+        indexToFocus = lastIndex;
+      }
+    } else {
+      if (e.key === keys.ARROW_DOWN) {
+        indexToFocus = currentIndex + 1;
+        if (indexToFocus > lastIndex) {
+          indexToFocus = 0;
+        }
+      } else if (e.key === keys.ARROW_UP) {
+        indexToFocus = currentIndex - 1;
+        if (indexToFocus < 0) {
+          indexToFocus = lastIndex;
+        }
+      }
+    }
+
+    actionsRef.current[indexToFocus!].focus();
+  }, []);
+
+  return {
+    panelRef,
+    panelProps: { onKeyDown },
+    popoverScreenReaderText: (
+      <EuiI18n
+        token="euiDataGridHeaderCell.actionsPopoverScreenReaderText"
+        default="To navigate through the list of column actions, press the Tab or Up/Down arrow keys."
```

Comment:
I have one question / nitpicky suggestion. VO + Firefox read "Up/down" really fast and it was tough to catch that meant I could use up and down arrows, on first listen. This may not be a problem, but what do you think about changing this to say "...press the Tab or Up and Down arrow keys." ?

Reply:
For sure!! I love fixing screen reader grammar nuances like these!
```suggestion
        default="To navigate through the list of column actions, press the Tab or Up and Down arrow keys."
```